packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -3,53 +3,50 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 
-import type {Path} from 'types/Config';
-import type {
+import {Config} from '@jest/types';
+import {
   DuplicatesSet,
   HTypeValue,
   ModuleMetaData,
   RawModuleMap,
   ModuleMapData,
   DuplicatesIndex,
   MockData,
-} from 'types/HasteMap';
+} from './types';
 
 import * as fastPath from './lib/fast_path';
 import H from './constants';
 
-const EMPTY_OBJ = {};
+const EMPTY_OBJ = {} as {[key: string]: any};
 const EMPTY_MAP = new Map();
 
-export opaque type SerializableModuleMap = {
+type ValueType<T> = T extends Map<T, infer V> ? V : T;
```

Comment:
This should be:
```suggestion
type ValueType<T> = T extends Map<unknown, infer V> ? V : T;
```
T is a Map in this case